src/Util/Examples/Flag.cs

```diff
@@ -0,0 +1,267 @@
+﻿// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Reflection;
+using SystemType = System.Type;
+
+
+namespace Google.Ads.GoogleAds.Examples
+{
+    /// <summary>
+    /// Base class for a command line flag.
+    /// </summary>
+    internal abstract class Flag
```

Comment:
Is .NET standard lib missing Flag library support? I wonder why a custom Flag implementation is necessary.

Reply:
Yes, it is missing one. There's Mono.Options library, but that assembly is not signed, which prevents me from using it in my project since our library is signed (and hence all dependencies have to be signed).